lib/lightstep/tracer/client_tracer.rb

```diff
@@ -228,204 +213,168 @@ def flush
-    # args.shift
-    # text = vsprintf(fmt, allArgs)
-    text = args.join(',')
-
-    self.raw_log_record({:level => level, :message => text}, args)
-
-    self.flush_if_needed
-    return text
+  # Internal use only.
+  #
+  # Generates a random ID (not a *true* UUID).
+  def generate_uuid_string
+    @tracer_utils.generate_guid
   end
 
-    # ==============================================================
-    # Internal use only.
-    # ==============================================================
-    def raw_log_record(fields, payload_array)
-        unless (@tracer_enabled)
-          return
-        end
+  # Internal use only.
+  def _finish_span(span)
+    return unless @tracer_enabled
+    span.set_end_micros(@tracer_utils.now_micros)
+    full = push_with_max(@tracer_span_records, span.to_thrift, @tracer_options[:max_span_records])
+    @tracer_counters[:dropped_spans] += 1 if full
+    flush_if_needed
+  end
 
-        fields[:runtime_guid] = @tracer_guid.to_s
+  # ==============================================================
+  # Internal use only.
+  # ==============================================================
+  def raw_log_record(fields, payload_array)
+    return unless @tracer_enabled
 
-        if (fields[:timestamp_micros].nil?)
-            fields[:timestamp_micros] = @tracer_utils.now_micros.to_i
-        end
+    fields[:runtime_guid] = @tracer_guid.to_s
 
-        # TODO: data scrubbing and size limiting
-        if (!payload_array.nil? && payload_array.size > 0)
-            json = JSON.generate(payload_array)
-            if (json.class.name == 'String')
-                fields[:payload_json] = json
-            end
-        end
+    if fields[:timestamp_micros].nil?
+      fields[:timestamp_micros] = @tracer_utils.now_micros.to_i
+    end
 
-        rec = LogRecord.new(fields)
-        full = self.push_with_max(@tracer_log_records, rec, @tracer_options[:max_log_records])
-        if full
-            @tracer_counters[:dropped_logs] += 1
-        end
+    # TODO: data scrubbing and size limiting
+    if !payload_array.nil? && !payload_array.empty?
+      json = JSON.generate(payload_array)
+      fields[:payload_json] = json if json.class.name == 'String'
     end
 
+    rec = LogRecord.new(fields)
+    full = push_with_max(@tracer_log_records, rec, @tracer_options[:max_log_records])
+    @tracer_counters[:dropped_logs] += 1 if full
+  end
+
   protected
 
-    def push_with_max(arr, item, max)
-      unless (max > 0)
-        max = 1
-      end
+  def push_with_max(arr, item, max)
+    max = 1 unless max > 0
 
-      arr << item
+    arr << item
 
-      # Simplistic random discard
-      count = arr.size
-      if (count > max)
-        i = @tracer_utils.randIntRange(0, max - 1)
-        arr[i] = arr.pop
-        return true
-      else
-        return false
-      end
+    # Simplistic random discard
+    count = arr.size
+    if count > max
+      i = rand(0..(max - 2)) # rand(a..b) is inclusive
```

Comment:
`randIntRange` was not defined and has been replaced with a standard `rand` call